### PR TITLE
Validate Kubernetes Scripts

### PR DESCRIPTION
**Pull request by ksubrmnn**
@madhanrm

### CODE REVIEW

File: Kubernetes/windows/debug/ValidateKubernetesHelper.psm1 [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    return $containers;
+}
+
+function TestConnectivity()
+{
+    param(
+        [string] $containerName,
+        [string] $remoteHost =  "172.217.3.206", #google.com
```
Comment (madhanrm) [inline Kubernetes/windows/debug/ValidateKubernetesHelper.psm1]: Doens;t name work here? That would validate DNS as well

File: Kubernetes/windows/debug/ValidateKubernetesHelper.psm1 [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    throw "TCP connection to ${remoteHost}:${port} failed from $containerName."
+}
+
+function GetContainerIPv4Address()
+{
+    param(
+        [string] $containerName
+    )
+
+    $matches = (kubectl exec $containerName ipconfig | Out-String  | Select-String -Pattern '(?sm)(IPv4 Address).*?: (.*?)\r\n' -AllMatches).Matches
```
Comment (madhanrm) [inline Kubernetes/windows/debug/ValidateKubernetesHelper.psm1]: can you also add a test to validate if kubectl get pod <podname> is reporting the same podip as the container ip?

**Comment by madhanrm** [comment]
This is cool. I'll merge this and we can add more tests.

**Comment by madhanrm** [comment]
This is cool. I'll merge this and we can add more tests.

can you move the files under windows/test  instead of windows/debug?

**Comment by daschott** [comment]
Awesome, thanks for doing this. I will try it out and see if it makes sense to add to next docs update. I'm sure it would be useful for users doing validation as well.

File: Kubernetes/windows/debug/ValidateKubernetesHelper.psm1 (hunk 2 above)
Comment (ksubrmnn) [reply Kubernetes/windows/debug/ValidateKubernetesHelper.psm1]: This test compares ipconfig with kubectl reported ip

**Comment by ksubrmnn** [comment]
@madhanrm moved the files. I'll include a DNS check when I use the updated CNI and follow up in a separate PR

**Pull request merged**